consensus/blake3pow/consensus.go

```diff
@@ -281,34 +283,29 @@ func (blake3pow *Blake3pow) verifyHeader(chain consensus.ChainHeaderReader, head
 	return nil
 }
 
-// CalcDifficulty is the difficulty adjustment algorithm. It returns
-// the difficulty that a new block should have when created at time
-// given the parent block's time and difficulty.
-func (blake3pow *Blake3pow) CalcDifficulty(chain consensus.ChainHeaderReader, time uint64, parent *types.Header) *big.Int {
-	return CalcDifficulty(chain.Config(), time, parent)
-}
-
-// CalcDifficultyAtIndex is the difficulty adjustment algorithm. It returns
-// the difficulty that a new block should have when created at time
-// given the parent block's time and difficulty at a given context.
-func (blake3pow *Blake3pow) CalcDifficultyAtIndex(chain consensus.ChainHeaderReader, time uint64, parent *types.Header, context int) *big.Int {
-	return CalcDifficultyAtIndex(chain.Config(), time, parent, context)
-}
-
 // CalcDifficulty is the difficulty adjustment algorithm. It returns
 // the difficulty that a new block should have when created at time
 // given the parent block's time and difficulty.
 // NOTE: This is essentially the Ethereum DAA, without a 'difficulty bomb'
-func CalcDifficulty(config *params.ChainConfig, time uint64, parent *types.Header) *big.Int {
+func (blake3pow *Blake3pow) CalcDifficulty(chain consensus.ChainHeaderReader, parent *types.Header) *big.Int {
 	nodeCtx := common.NodeLocation.Context()
+
 	// https://github.com/ethereum/EIPs/issues/100.
 	// algorithm:
 	// diff = (parent_diff +
 	//         (parent_diff / 2048 * max((2 if len(parent.uncles) else 1) - ((timestamp - parent.timestamp) // 9), -99))
 	//        ) + 2^(periodCount - 2)
 
+	time := parent.Time()
+
+	if parent.Hash() == chain.Config().GenesisHash {
+		return parent.Difficulty()
+	}
+
+	parentOfParent := chain.GetHeaderByHash(parent.ParentHash())
+
 	bigTime := new(big.Int).SetUint64(time)
-	bigParentTime := new(big.Int).SetUint64(parent.Time())
+	bigParentTime := new(big.Int).SetUint64(parentOfParent.Time())
```

Comment:
A) I don't think the DAA changes belong in this PR
B) I believe this causes an error when dT is 0

Reply:
We need this to actually run the new core. I can get it as a separate PR if needed.

Reply:
I think that this should be added here as it doesn't run without.

Reply:
Oh yes, in that case, fine to bring it in this PR :)

Reply:
But, what happens when `bigTime == bigParentTime`?

Reply:
No problem. There is no division by zero.